pkg/serializer/internal/metrics/iterable_series.go

```diff
@@ -500,17 +500,35 @@ func (pb *PayloadsBuilder) finishPayload() error {
 // FIXME(maxime): to be removed when v2 endpoints are available
 func (series *IterableSeries) MarshalJSON() ([]byte, error) {
 	// use an alias to avoid infinite recursion while serializing a Series
-	type SeriesAlias Series
+	type SeriesAlias struct {
+		*metrics.Serie
+		Metadata map[string]interface{} `json:"metadata,omitempty"`
+	}
 
-	seriesAlias := make(SeriesAlias, 0)
+	seriesAlias := make([]SeriesAlias, 0)
 	for series.MoveNext() {
 		serie := series.source.Current()
 		serie.PopulateDeviceField()
 		serie.PopulateResources()
-		seriesAlias = append(seriesAlias, serie)
+
+		serieWithMetadata := SeriesAlias{
+			Serie: serie,
+		}
+
+		if serie.Source != 0 {
+			serieWithMetadata.Metadata = map[string]interface{}{
+				"origin": map[string]interface{}{
+					"origin_product":        metricSourceToOriginProduct(serie.Source),
+					"origin_sub_product":    metricSourceToOriginCategory(serie.Source),
+					"origin_product_detail": metricSourceToOriginService(serie.Source),
```

Comment:
Can we also figure out a mapping for product detail? We need to set product detail to something which helps us identify whether it's an enhanced/runtime/standard (non-billable) or custom (billable) metric. This mapping will live in https://github.com/DataDog/dd-analytics/blob/master/luigiscripts/billing/usage/standard_metric_allowlist.json and help inform billing decisions in the future along with metric prefix.

Reply:
https://github.com/DataDog/dd-source/pull/208062 - I'll make a small follow-up PR to map the details properly